demos/http/http_demo_s3_download/http_demo_s3_download.c

```diff
@@ -352,10 +1021,91 @@ static bool downloadS3ObjectFile( const TransportInterface_t * pTransportInterfa
         }
         else
         {
-            LogError( ( "Failed to initialize HTTP request headers: Error=%s.",
+            LogError( ( "Failed to add range header to request headers: Error=%s.",
                         HTTPClient_strerror( httpStatus ) ) );
         }
 
+        /* Get the hash of the payload. */
+        sha256( ( const char * ) "", 0, pPayloadHashDigest );
+        lowercaseHexEncode( ( const char * ) pPayloadHashDigest, strlen( pPayloadHashDigest ), hexencoded );
+
+        if( returnStatus == true )
+        {
+            httpStatus = HTTPClient_AddHeader( &requestHeaders,
+                                               ( const char * ) SIGV4_HTTP_X_AMZ_CONTENT_SHA256_HEADER,
+                                               ( size_t ) ( sizeof( SIGV4_HTTP_X_AMZ_CONTENT_SHA256_HEADER ) - 1 ),
+                                               ( const char * ) hexencoded,
+                                               64 );
+
+            if( httpStatus != HTTPSuccess )
+            {
+                LogError( ( "Failed to add X-AMZ-CONTENT-SHA256-HEADER to request headers: Error=%s.",
+                            HTTPClient_strerror( httpStatus ) ) );
+                returnStatus = false;
+            }
+        }
+
+        /* Move request header pointer past the initial headers which are added by coreHTTP
+         * library and are not required by SigV4 library. */
+        pHeaders = ( const char * ) requestHeaders.pBuffer;
+        headersLen = requestHeaders.headersLen;
+
+        while( 0 != strncmp( pHeaders, "\r\n", strlen( "\r\n" ) ) )
+        {
+            pHeaders++;
+            headersLen--;
+        }
+
+        /* Moving header pointer past "\r\n" .*/
+        pHeaders = pHeaders + 2;
+        headersLen = headersLen - 2;
+
+        /* Setup the HTTP parameters. */
+        sigv4HttpParams.pHttpMethod = requestInfo.pMethod;
+        sigv4HttpParams.httpMethodLen = requestInfo.methodLen;
+        /* None of the requests parameters below are pre-canonicalized */
+        sigv4HttpParams.flags = 0;
+        sigv4HttpParams.pPath = requestInfo.pPath;
+        sigv4HttpParams.pathLen = requestInfo.pathLen;
+        sigv4HttpParams.pQuery = NULL;
+        sigv4HttpParams.queryLen = 0;
+        sigv4HttpParams.pHeaders = pHeaders;
+        sigv4HttpParams.headersLen = headersLen;
+        sigv4HttpParams.pPayload = "";
```

Comment:
Once the macro is created for the empty payload
```suggestion
        sigv4HttpParams.pPayload = S3_REQUEST_EMPTY_PAYLOAD ;
```